File: 백준/Silver/2839. 설탕 배달/설탕 배달.js
const fs = require('fs');

// 백준 제출용
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const n = Number(input);

let minVinyl = 5001;

for (let i = 0; i <= Math.floor(n / 5); i++) {
  const r = n - 5 * i;

  if (r % 3 === 0) {
    const rr = r / 3;
    minVinyl = Math.min(minVinyl, i + rr);
  }
}

console.log(minVinyl === 5001 ? -1 : minVinyl);

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {
	public static final int M = Integer.MAX_VALUE;

	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int minVinyl = 5001;
		for(int i = 0, size = n/5; i <= size; i++) {
			int r = n - 5*i;
			if(r % 3 == 0) {
				minVinyl = Math.min(minVinyl, i + r/3);
			}
		}
		System.out.println(minVinyl == 5001 ? -1 : minVinyl
		);
	} // end of main
} // end of class